knxdclient/client.py

```diff
@@ -113,10 +113,11 @@ async def connect(self, host: str = 'localhost', port: int = 6720, sock: Optiona
 
 
     async def _read_raw_knxpacket(self) -> bytes:
-        reader = self._reader
-        assert reader is not None
-        length = int.from_bytes(await reader.readexactly(2), byteorder='big')
-        return await reader.readexactly(length)
+        if self._reader is None or self._reader.at_eof():
+            raise ConnectionError("No connection to KNXD has been established yet or the previous connection's "
+                                  "StreamReader is at EOF")
```

Comment:
Does this work when manually closing the connection via `close()` such that no exception is raised in this case?